smart-wallets/test/scenario/YieldDistributionToken.t.sol

```diff
@@ -7,9 +7,45 @@ import { Test } from "forge-std/Test.sol";
 
 import { Deposit, UserState } from "../../src/token/Types.sol";
 import { YieldDistributionTokenHarness } from "../harness/YieldDistributionTokenHarness.sol";
+import { stdError } from "forge-std/StdError.sol";
+
+import { console2 } from "forge-std/console2.sol";
 
 contract YieldDistributionTokenScenarioTest is Test {
 
+    /**
+     * @notice Emitted when yield is deposited into the YieldDistributionToken
+     * @param user Address of the user who deposited the yield
+     * @param currencyTokenAmount Amount of CurrencyToken deposited as yield
+     */
+    event Deposited(address indexed user, uint256 currencyTokenAmount);
+
+    /**
+     * @notice Emitted when yield is claimed by a user
+     * @param user Address of the user who claimed the yield
+     * @param currencyTokenAmount Amount of CurrencyToken claimed as yield
+     */
+    event YieldClaimed(address indexed user, uint256 currencyTokenAmount);
+
+    /**
+     * @notice Emitted when yield is accrued to a user
+     * @param user Address of the user who accrued the yield
+     * @param currencyTokenAmount Amount of CurrencyToken accrued as yield
+     */
+    event YieldAccrued(address indexed user, uint256 currencyTokenAmount);
+
+    // Errors
+
+    /**
+     * @notice Indicates a failure because the transfer of CurrencyToken failed
+     * @param user Address of the user who tried to transfer CurrencyToken
+     * @param currencyTokenAmount Amount of CurrencyToken that failed to transfer
+     */
+    error TransferFailed(address user, uint256 currencyTokenAmount);
+
+    /// @notice Indicates a failure because a yield deposit is made in the same block as the last one
+    error DepositSameBlock();
```

Comment:
For all of these errors, we can `import { YieldDistributionToken } from "../../src/token/YieldDistributionToken.sol"` and then `vm.expectRevert(YieldDistributionToken.DepositSameBlock.selector)`